### PR TITLE
#164797147 Ft Enable Google authentication

### PR DESCRIPTION
**Pull request by musonant**
#### What does this PR do?
* Setup google authentication

#### Description of Task to be completed?
* Write tests to test google authentication
* Setup google with passport

#### How should this be manually tested?
* After cloning repo and starting server
* On your browser, attempt to open http:localhost:3000/api/v1/auth/twitter

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?(if applicable)
#164797147

#### Screenshots
Screenshot 2019-04-25 at 6 30 22 PM
Screenshot 2019-04-25 at 6 30 43 PM

### CODE REVIEW

File: src/db/migrations/20190325172457-create-users.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export default {
       },
       password: {
         type: Sequelize.STRING,
-        allowNull: false,
+        allowNull: true,
```
Comment (emmsdan) [inline src/db/migrations/20190325172457-create-users.js:32]: Please, I need clarity on why you changed `allowNull`  to `TRUE` and does it mean that a user can be created without providing a password.

Comment (musonant) [reply src/db/migrations/20190325172457-create-users.js:32]: Exactly. When a user registers with a third-party application like Facebook, his password would not be provided by Facebook. However, he is able to create/edit password on his profile.

Comment (emmsdan) [reply src/db/migrations/20190325172457-create-users.js:32]: Oh, I get it now. Nice one

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 language: node_js
 node_js:
-  - "stable"
+  - '11.11.0'
```
Comment (Japhethca) [inline .travis.yml]: can we use an LTS node version here?

File: src/tests/routes/controllers/social-auth.test.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import { socialController } from '../../../routes/controllers/auth.controller';
+import {
+  facebookAuth,
+  twitterAuth,
+  googleAuth,
+} from '../../../routes/services/passport-strategies.services';
+
+chai.use(chaiHttp);
+
+describe('', () => {
```
Comment (Japhethca) [inline src/tests/routes/controllers/social-auth.test.js]: There should be a description or title for this describe block.

**Review by Japhethca — COMMENTED** [review]
Nice work! Just a few comment.

File: .travis.yml (hunk 2 above)
Comment (musonant) [reply .travis.yml]: Yes. I have updated this PR @Japhethca 
Thanks.

**Review by Japhethca — APPROVED** [review]

**Pull request merged**